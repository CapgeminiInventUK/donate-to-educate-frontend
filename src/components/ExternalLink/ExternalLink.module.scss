@use '@assets/global.scss' as g;

.link {
  color: g.$secondary-mid-blue;
  font-size: 16px;
  font-style: normal;
  font-weight: 500;
  line-height: 160%;
}
