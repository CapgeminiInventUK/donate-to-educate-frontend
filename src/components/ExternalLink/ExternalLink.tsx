import type { ExternalLinkProps } from '@/types/props';
import type { FC } from 'react';
import styles from './ExternalLink.module.scss';

const ExternalLink: FC<ExternalLinkProps> = ({ linkText, linkUrl, ariaLabel, className }) => {
  return (
    <a
      className={`${styles.link} ${className}`}
      href={linkUrl}
      target="_blank"
      rel="noreferrer"
      aria-label={ariaLabel}
    >
      {linkText}
    </a>
  );
};
export default ExternalLink;
