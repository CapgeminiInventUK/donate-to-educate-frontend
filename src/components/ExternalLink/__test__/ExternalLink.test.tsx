import { render } from '@testing-library/react';
import ExternalLink from '../ExternalLink';

describe('Internal link', () => {
  it('should render component', () => {
    const screen = render(
      <ExternalLink linkText={'linkText'} linkUrl={'linkUrl'} ariaLabel={'aria-label'} />
    );
    expect(screen).toMatchSnapshot();
  });
});
