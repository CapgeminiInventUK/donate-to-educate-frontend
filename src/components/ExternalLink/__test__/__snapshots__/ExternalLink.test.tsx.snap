// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Internal link > should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="aria-label"
        class="_link_fa3312 undefined"
        href="linkUrl"
        rel="noreferrer"
        target="_blank"
      >
        linkText
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="aria-label"
      class="_link_fa3312 undefined"
      href="linkUrl"
      rel="noreferrer"
      target="_blank"
    >
      linkText
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
